File: BidWinKoModify/app/src/main/java/com/bidwinko/screens/activity/BuyBidActivity.java
package com.bidwinko.screens.activity;

import android.app.Activity;
import android.app.ProgressDialog;
import android.content.Intent;
import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.text.Html;
import android.util.Log;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.bidwinko.R;
import com.bidwinko.adapter.BuyBidListAdapter;
import com.bidwinko.API.APIService;
import com.bidwinko.model.BuyBid;
import com.bidwinko.model.BuyBidValue;
import com.bidwinko.utilies.Constants;
import com.bidwinko.API.Retrofit;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class BuyBidActivity extends AppCompatActivity {

    RecyclerView recyclerView;
    ProgressDialog progressDialog;
    ArrayList<BuyBidValue> buybidArraylists = new ArrayList<>();
    private BuyBidListAdapter mAdapter;
    TextView txtBidBalance;
    CardView unlimitedBid;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.buybid_fragment);

        getSupportActionBar().setHomeButtonEnabled(true);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        final Drawable upArrow = getResources().getDrawable(R.drawable.ic_arrow_back);
        upArrow.setColorFilter(getResources().getColor(R.color.white), PorterDuff.Mode.SRC_ATOP);
        getSupportActionBar().setHomeAsUpIndicator(upArrow);
        ((AppCompatActivity)this).getSupportActionBar().setTitle((Html.fromHtml("<font color=\"#ffffff\">" + getString(R.string.buybid) + "</font>")));

        recyclerView = findViewById(R.id.recycler_view);
        txtBidBalance = findViewById(R.id.bidbalance);
        unlimitedBid = findViewById(R.id.card_unlimited_bid);

//        getBuyBids();

        unlimitedBid.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
//				Toast.makeText(getActivity(),"Feature Coming Soon..",Toast.LENGTH_SHORT).show();
                Intent intent = new Intent(BuyBidActivity.this,ShareEarnActivity.class);
                startActivity(intent);
            }
        });


    }

//    private void getBuyBids() {
//
//        APIService apiService = Retrofit.getClient().create(APIService.class);
//        Call<BuyBid> call = apiService.getbuyBids(Constants.getSharedPreferenceInt(BuyBidActivity.this,"userId",0),
//                Constants.getSharedPreferenceString(BuyBidActivity.this,"securitytoken",""),
//                Constants.getSharedPreferenceString(BuyBidActivity.this,"versionName",""),
//                Constants.getSharedPreferenceInt(BuyBidActivity.this,"versionCode",0),
//                Constants.getSharedPreferenceString(BuyBidActivity.this,"userFrom",""));
//
//        if(!((Activity) BuyBidActivity.this).isFinishing()) {
//            progressDialog = new ProgressDialog(BuyBidActivity.this);
//            progressDialog.setMessage(getString(R.string.loadingwait));
//            progressDialog.show();
//            progressDialog.setCancelable(false);
//        }
//
//        call.enqueue(new Callback<BuyBid>() {
//            @Override
//            public void onResponse(Call<BuyBid>call, Response<BuyBid> response) {
//                dismissProgressDialog();
////                Log.d("Bid", "Number of movies received:size "+response +" bid");
//
//                if(response!=null){
//                    if(response.isSuccessful()){
//                        if(response.body().getStatus()==200){
////                            txtBidBalance.setText(""+response.body().getBidBal());
////                            boolean res = response.body().getFreeBid();
//                            String bidremain = response.body().getBidBalance();
//                            Constants.setSharedPreferenceString(BuyBidActivity.this,"bidBalance",bidremain);
//                            txtBidBalance.setText(""+Constants.getSharedPreferenceString(BuyBidActivity.this,"bidBalance",""));
//                            buybidArraylists = response.body().getBuyBidValues();
////                            Log.d("Bid", "Number of movies received:size " + buybidArraylists.size());
//
////                            boolean walletOption = response.body().getWalletOpt();
//                            mAdapter = new BuyBidListAdapter(response,buybidArraylists, BuyBidActivity.this);
//                            RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(BuyBidActivity.this);
//                            recyclerView.setLayoutManager(mLayoutManager);
//                            recyclerView.setItemAnimator(new DefaultItemAnimator());
//                            recyclerView.setAdapter(mAdapter);
//                        }else{
//                            Toast.makeText(BuyBidActivity.this,getString(R.string.systemmessage)+response.body().getMessage(),Toast.LENGTH_SHORT).show();
//                        }
//
//                    }
//                }
//                else{
//                    Toast.makeText(BuyBidActivity.this,getString(R.string.systemmessage)+response.errorBody(),Toast.LENGTH_SHORT).show();
//                }
//
//
//            }
//
//            @Override
//            public void onFailure(Call<BuyBid>call, Throwable t) {
//                // Log error here since request failed
//                Log.e("response", t.toString());
//            }
//        });
//
//
//    }

    private void dismissProgressDialog() {
        if (progressDialog != null && progressDialog.isShowing()) {
            progressDialog.dismiss();
        }
    }

    @Override
    public void onDestroy() {
        dismissProgressDialog();
        super.onDestroy();
    }

    @Override
    public void onPause() {
        dismissProgressDialog();
        super.onPause();
    }


    @Override
    public void onBackPressed() {
        super.onBackPressed();
        this.finish();


    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if(requestCode==103){
            finish();
            startActivity(getIntent());
        }
    }

}

File: BidWinKoModify/app/src/main/java/com/bidwinko/screens/activity/BuyBidActivity.kt
package com.bidwinko.screens.activity

import android.animation.Animator
import android.animation.Animator.AnimatorListener
import android.app.Activity
import android.app.ProgressDialog
import android.content.Intent
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.text.Html
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.bidwinko.R
import com.bidwinko.adapter.BuyBidPlanAdapter
import com.bidwinko.databinding.BuybidFragmentBinding
import com.bidwinko.model.RequestModels.CommonRequest
import com.bidwinko.model.ResponseModels.BidPlan
import com.bidwinko.utilies.Constants
import com.bidwinko.utilies.SessionManager
import com.bidwinko.viewModel.mainViewModel

class BuyBidActivity : AppCompatActivity() {
    var progressDialog: ProgressDialog? = null
    private var mAdapter: BuyBidPlanAdapter? = null
    var txtBidBalance: TextView? = null
    var unlimitedBid: CardView? = null

    lateinit var binding : BuybidFragmentBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = BuybidFragmentBinding.inflate(layoutInflater)
        setContentView(binding.root)
        supportActionBar!!.setHomeButtonEnabled(true)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setBackgroundDrawable(ColorDrawable(resources.getColor(R.color.black)))
        val upArrow = getResources().getDrawable(R.drawable.ic_arrow_back)
        supportActionBar!!.setHomeAsUpIndicator(upArrow)
        (this as AppCompatActivity).supportActionBar!!.setTitle(
            Html.fromHtml(
                "<font color=\"#ffffff\">" + getString(
                    R.string.buybid
                ) + "</font>"
            )
        )
        txtBidBalance = findViewById(R.id.bidbalance)
        unlimitedBid = findViewById(R.id.card_unlimited_bid)
        binding.bidbalance.text = SessionManager(this).GetValue(Constants.TOTAL_BIDS)

        binding.buybidrefresh.setOnRefreshListener {
            binding.shimmer.root.startShimmer()
            binding.shimmer.root.visibility = View.VISIBLE
            binding.bidRecyclerView.visibility = View.GONE
            getData()
        }
        binding.cardUnlimitedBid.setOnClickListener {
            val intentbuybid = Intent(this, ReferActivity::class.java)
            startActivity(intentbuybid)
        }

        binding.choosetv.setOnClickListener {
            val intentbuybid = Intent(this, ReferActivity::class.java)
            startActivity(intentbuybid)
        }


        binding.anim.let {
            it.playAnimation()
            it.addAnimatorListener(object : AnimatorListener{
                override fun onAnimationStart(animation: Animator) {}
                override fun onAnimationEnd(animation: Animator) {
                    it.playAnimation()
                }
                override fun onAnimationCancel(animation: Animator) {}
                override fun onAnimationRepeat(animation: Animator) {}
            })
        }
        getData()
    }

    private fun getData() {

        val request = CommonRequest(
            userId =  SessionManager(this).GetValue(Constants.USER_ID).toString(),
            securityToken = SessionManager(this).GetValue(Constants.SECURITY_TOKEN),
            versionName = SessionManager(this).GetValue(Constants.VERSION_NAME),
            versionCode = SessionManager(this).GetValue(Constants.VERSION_CODE)
        )
        mainViewModel(this).GetBidsPacakage(request).observe(this){
            binding.buybidrefresh.isRefreshing = false
            if(it.status == 200){
                SessionManager(this).InitializeValue(Constants.TOTAL_BIDS,it.totalBids)
                binding.bidbalance.text = SessionManager(this).GetValue(Constants.TOTAL_BIDS)
                    mAdapter = BuyBidPlanAdapter(it.bidPlans as ArrayList<BidPlan>, this@BuyBidActivity)
                binding.bidRecyclerView.setAdapter(mAdapter)
                binding.shimmer.root.stopShimmer()
                binding.shimmer.root.visibility = View.GONE
                binding.bidRecyclerView.visibility = View.VISIBLE
            }
        }
    }

    public override fun onDestroy() {
        super.onDestroy()
    }

    public override fun onPause() {
        super.onPause()
    }

    override fun onBackPressed() {
        super.onBackPressed()
        finish()
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == 103) {
            finish()
            startActivity(intent)
        }
    }
}
